File: LC50/lc50.py
"""
Implement pow(x, n)
"""


def mypow(base, power):
    if power == 0:
        return 1
    if power < 0:
        power = -power
        base = 1 / base
    if power % 2 == 0:
        return mypow(base * base, power // 2)
    return mypow(base * base, power // 2) * base


print(pow(3, 2))
print(pow(3, -2))
print(pow(8.88023, 3))

File: LC50/lc50.js
function pow(x, n) {
    if (n === 0) {
        return 1;
    }
    if (n < 0) {
        n = -n;
        x = 1 / x;
    }
    if (n % 2 === 0) {
        return pow(x*x, Math.floor(n/2));
    } else {
        return x * pow(x*x, Math.floor(n/2));
    }
}

console.log(pow(3, 2))
console.log(pow(3, -2))
console.log(pow(8.88023, 3))
